src/libsync/discovery.cpp

```diff
@@ -368,6 +368,29 @@ void ProcessDirectoryJob::processFileAnalyzeRemoteInfo(
     item->_directDownloadUrl = serverEntry.directDownloadUrl;
     item->_directDownloadCookies = serverEntry.directDownloadCookies;
 
+    // Check for missing server data
+    if (serverEntry.size == -1
```

Comment:
I would remove this `if`, and add a condition later: `if (!missingData.isEmpty())`,  so one avoids to repeat every condition.
(But that's just cosmetic)

Reply:
Yep